### DISCUSSION TITLE
Changing node IDs for organizations and repos

### DISCUSSION THREAD

**Discussion by weineran** [community/community#35719] [Apps, API and Webhooks]
Over the past few days, we've noticed that the `node_id` for several organizations and repositories have changed a couple times.

For example, currently our organization has a `node_id` of `MDEyOk9yZ2FuaXphdGlvbjkwMzk1ODUx`:
```
$ curl   -H "Accept: application/vnd.github+json"   -H "Authorization: Bearer $GITHUB_TOKEN"   -s https://api.github.com/orgs/gitleaks | jq --raw-output '.node_id'
MDEyOk9yZ2FuaXphdGlvbjkwMzk1ODUx
```

However, at some point between 10/5/2022 and today (sorry I don't have exact timestamps), that same API call returned a different node_id for the same org: `O_kgDOBWNUyw`

Similarly, repo node_ids have changed as well. One of the repos within that organization currently has a `node_id` of `MDEwOlJlcG9zaXRvcnkyNDI4NTQ5MDk=`
```
$ curl   -H "Accept: application/vnd.github+json"   -H "Authorization: Bearer $GITHUB_TOKEN"   -s   https://api.github.com/repos/gitleaks/gitleaks-action | jq --raw-output '.node_id'
MDEwOlJlcG9zaXRvcnkyNDI4NTQ5MDk=
```

But at some point between 10/5/2022 and today, that same API call returned a different node_id for the same repo: `R_kgDODnmr_Q`

We have noticed similar changes over the past few days with several of our customers' org/repo node_ids.

I suppose this means that we cannot rely on node_ids remaining static, but I can't find anywhere in the GitHub docs stating this explicitly. Does anyone know if node_ids are expected to change like this?

**Comment by weineran** [comment]
This is a similar question where somebody noticed that the `id` field of a repo changed: https://github.com/orgs/community/discussions/29784

**Reply by LangLangBart** [reply]
This seems to be the new `next ID`

```zsh
gh api /orgs/gitleaks --jq '.node_id'
# MDEyOk9yZ2FuaXphdGlvbjkwMzk1ODUx


gh api -H "X-Github-Next-Global-ID: 1" /orgs/gitleaks --jq '.node_id'
# O_kgDOBWNUyw
```

---

> I suppose this means that we cannot rely on node_ids remaining static, but I can't find anywhere in the GitHub docs stating this explicitly. Does anyone know if node_ids are expected to change like this?

See this post:
- [GraphQL global ID migration update](https://github.blog/2021-11-16-graphql-global-id-migration-update/) (16/Nov/21)
